File: docs/breez-sdk/snippets/csharp/SeedlessRestore.cs
using Breez.Sdk.Spark;

namespace BreezSdkSnippets
{
    // ANCHOR: implement-prf-provider
    // In practice, implement PRF provider using platform passkey APIs
    class ExamplePasskeyPrfProvider : PasskeyPrfProvider
    {
        public async Task<byte[]> DerivePrfSeed(string salt)
        {
            // Call platform passkey API with PRF extension
            // Returns 32-byte PRF output
            throw new NotImplementedException("Implement using platform passkey APIs");
        }

        public async Task<bool> IsPrfAvailable()
        {
            // Check if PRF-capable passkey exists
            throw new NotImplementedException("Check platform passkey availability");
        }
    }
    // ANCHOR_END: implement-prf-provider

    class SeedlessRestoreSnippets
    {
        async Task<Seed> CreateSeed()
        {
            // ANCHOR: create-seed
            var prfProvider = new ExamplePasskeyPrfProvider();
            var seedless = new SeedlessRestore(prfProvider, null);

            // Create a new seed with user-chosen salt
            // The salt is published to Nostr for later discovery
            var seed = await seedless.CreateSeed(salt: "personal");

            // Use the seed to initialize the SDK
            var config = BreezSdkSparkMethods.DefaultConfig(network: Network.Mainnet);
            var builder = new SdkBuilder(config: config, seed: seed);
            await builder.WithDefaultStorage(storageDir: "./.data");
            var sdk = await builder.Build();
            // ANCHOR_END: create-seed
            return seed;
        }

        async Task<List<string>> ListSalts()
        {
            // ANCHOR: list-salts
            var prfProvider = new ExamplePasskeyPrfProvider();
            var seedless = new SeedlessRestore(prfProvider, null);

            // Query Nostr for salts associated with this passkey
            var salts = await seedless.ListSalts();

            foreach (var salt in salts)
            {
                Console.WriteLine($"Found wallet: {salt}");
            }
            // ANCHOR_END: list-salts
            return salts;
        }

        async Task<Seed> RestoreSeed()
        {
            // ANCHOR: restore-seed
            var prfProvider = new ExamplePasskeyPrfProvider();
            var seedless = new SeedlessRestore(prfProvider, null);

            // Restore seed using a known salt
            var seed = await seedless.RestoreSeed(salt: "personal");

            // Use the seed to initialize the SDK
            var config = BreezSdkSparkMethods.DefaultConfig(network: Network.Mainnet);
            var builder = new SdkBuilder(config: config, seed: seed);
            await builder.WithDefaultStorage(storageDir: "./.data");
            var sdk = await builder.Build();
            // ANCHOR_END: restore-seed
            return seed;
        }
    }
}

File: docs/breez-sdk/snippets/python/src/seedless_restore.py
# pylint: disable=duplicate-code
from breez_sdk_spark import (
    BreezSdk,
    PasskeyPrfProvider,
    SeedlessRestore,
    SdkBuilder,
    default_config,
    Network,
)


# ANCHOR: implement-prf-provider
# Implement PRF provider using platform passkey APIs
class ExamplePasskeyPrfProvider(PasskeyPrfProvider):
    def derive_prf_seed(self, salt: str) -> bytes:
        # Call platform passkey API with PRF extension
        # Returns 32-byte PRF output
        raise NotImplementedError("Implement using platform passkey APIs")

    def is_prf_available(self) -> bool:
        # Check if PRF-capable passkey exists
        raise NotImplementedError("Check platform passkey availability")
# ANCHOR_END: implement-prf-provider


async def create_seed() -> BreezSdk:
    # ANCHOR: create-seed
    prf_provider = ExamplePasskeyPrfProvider()
    seedless = SeedlessRestore(prf_provider, None)

    # Create a new seed with user-chosen salt
    # The salt is published to Nostr for later discovery
    seed = await seedless.create_seed(salt="personal")

    # Use the seed to initialize the SDK
    config = default_config(network=Network.MAINNET)
    builder = SdkBuilder(config=config, seed=seed)
    await builder.with_default_storage(storage_dir="./.data")
    sdk = await builder.build()
    # ANCHOR_END: create-seed
    return sdk


async def list_salts() -> list[str]:
    # ANCHOR: list-salts
    prf_provider = ExamplePasskeyPrfProvider()
    seedless = SeedlessRestore(prf_provider, None)

    # Query Nostr for salts associated with this passkey
    salts = await seedless.list_salts()

    for salt in salts:
        print(f"Found wallet: {salt}")
    # ANCHOR_END: list-salts
    return salts


async def restore_seed() -> BreezSdk:
    # ANCHOR: restore-seed
    prf_provider = ExamplePasskeyPrfProvider()
    seedless = SeedlessRestore(prf_provider, None)

    # Restore seed using a known salt
    seed = await seedless.restore_seed(salt="personal")

    # Use the seed to initialize the SDK
    config = default_config(network=Network.MAINNET)
    builder = SdkBuilder(config=config, seed=seed)
    await builder.with_default_storage(storage_dir="./.data")
    sdk = await builder.build()
    # ANCHOR_END: restore-seed
    return sdk
